### PR TITLE
✨ NEW: Add material design icons roles

### LINKED ISSUE #36: Is there a plan to add material design inline icons?
I thought I'd ask this here since you're moving away from sphinx-panels - this is looking great BTW!

I see that there are inline icons for FontAwesome and Octicon. I'm helping develop a port of mkdocs-material theme to sphinx, and our docs could use the set of inline icons from the material design set.

## For reference
1. This is the theme I'm helping: Its called [sphinx-immaterial](https://github.com/jbms/sphinx-immaterial) because there was a previous (kind of a "hobbyist" attempt) called sphinx-material. 
2. This is the [set of icons supported by all the material design docs themes](https://github.com/squidfunk/mkdocs-material/tree/master/material/.icons/material) (in svg form)

---
ps - If you're curious to see the sphinx-immaterial theme in action, it is [hosted on gh-pages](https://jbms.github.io/sphinx-immaterial/index.html). I've also made sure it works rather well with RTD builds 😉 .

### PR DESCRIPTION
**Pull request by 2bndy5**
resolves #36 

This PR add 5 new roles for using Google's Material Design Icons as inline text. Each role represents a different material design "flavor":
- `material-regular`
- `material-outlined`
- `material-sharp`
- `material-round`
- `material-twotone`

Amongst all these flavors, there is a total of 10662 new icons provided. Instead of displaying them all, I defer users to browse [Goggle's Material Design Icons showcase](fonts.google.com/icons).

Includes updates to
- docs
- CSS (minified and SCSS)
- tests (pre/post XML files)
   - For some reason the pytests about myst icons-material-design.xml fails. I suspect it has something to do with the bullets in the bullet list in the docs/snippets/myst/icons-material-design.txt or just a difference in LF vs CRLF (I develop in Windows - but also ran pytest in WSL ubuntu) 😕 

---
Since the feature is working with the JSON files I programmatically generated from the google/material-design-icons repo, I set up [a small repo](https://github.com/2bndy5/material-design-icons-json) to run a CI job that updates JSON files when needed.

### CODE REVIEW

**Comment by 2bndy5** [comment]
I still have no idea what's wrong with the pytest for Myst icons-material-design.xml. 

I've tried using `*` as the bullets in the list, but that didn't help. I've tried manually editing both (pre/post) xml to not include the `bullet="-"` in the `bulletlist` tag, but then it complains that it is missing. 🤔

I'm not sure what is going on with the line that is triggering the `AssertionError`
https://github.com/executablebooks/sphinx-design/blob/629b86805210d5f6c1b059997008b4052c295aba/tests/test_snippets.py#L102

File: docs/badges_buttons.md [hunk 1]
```diff
@@ -215,7 +215,7 @@ Material Design icons come as several flavors. Each flavor represents a differen
 
 Not all icons are available for each flavor, but most are. Instead of displaying the 10660+ icons here, you are encouraged to browse the available icons from the [Material Design Icons' showcase](https://fonts.google.com/icons) hosted by Google.
 
-- A regular icon: {material-regular}`thumbs_up_down`, some more text
+- A regular icon: {material-regular}`data_exploration`, some more text
```
Comment (2bndy5) [inline docs/badges_buttons.md:218]: I'm ok with this. Originally, I just wanted to demonstrate the differences between flavors of the same icon.

Comment (chrisjsewell) [reply docs/badges_buttons.md]: 👍

File: tests/test_snippets.py [hunk 2]
```diff
@@ -33,8 +33,11 @@ def test_snippets_rst(
     builder.src_path.joinpath("index.rst").write_text(content, encoding="utf8")
     write_assets(builder.src_path)
     builder.build()
+    pformat = builder.get_doctree("index").pformat()
+    # fixed in https://github.com/executablebooks/MyST-Parser/pull/465
+    pformat = pformat.replace('<bullet_list bullet="-">', "<bullet_list>")
     file_regression.check(
-        builder.get_doctree("index").pformat(),
+        pformat,
```
Comment (2bndy5) [inline tests/test_snippets.py:40]: I'm glad to see this wasn't a fault on my end (unless I'm misunderstanding this change).

Comment (chrisjsewell) [reply tests/test_snippets.py:40]: No just a minor difference between RST and MyST AST

**Review by chrisjsewell — APPROVED** [review]
Nice one cheers!

**Comment by 2bndy5** [comment]
I may submit updates (via PR) to the compiled jsons when I see significant changes to the material icons repo. These updates may come without a linked issue though.

**Comment by chrisjsewell** [comment]
Sounds good ta 👍

**Comment by 2bndy5** [comment]
## Any idea about ETA for next release?
I'm sorry this sounds impatient (there's really no need to rush), but I keep teasing this feature over on the sphinx-immaterial theme repo, so I'm very much looking forward to the next release. We're starting to consider an optional extra CSS that exports user-specified SVG's data (based on node.js pkgs - not google's src repo like this feature is) as CSS vars...

**Comment by chrisjsewell** [comment]
No that's fair, sorry I got side-tracked with other things, but will try to get this done within the week

**Pull request merged**